frontend/pkg/frontend/middleware_tracing.go

```diff
@@ -16,24 +16,45 @@ import (
 	"github.com/Azure/ARO-HCP/internal/api/arm"
 )
 
-// MiddlewareTracing starts a span wrapping all incoming HTTP requests.
-// Other middlwares or actual request handlers can extend its metadata or create
-// their own associated spans.
+// MiddlewareTracing starts an OpenTelemetry span wrapping all incoming HTTP
+// requests. Other middlewares or actual request handlers can extend its
+// metadata or create their own associated spans.
+// The middleware expects that the trace provider is initialized and configured
+// in advance.
 func MiddlewareTracing(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
-	otelhttp.NewHandler(http.Handler(next), fmt.Sprintf("HTTP %s", r.Method)).ServeHTTP(w, r)
+	otelhttp.NewHandler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var (
+			ctx    = r.Context()
+			logger = LoggerFromContext(ctx)
+		)
+
+		data, err := CorrelationDataFromContext(ctx)
+		if err != nil {
+			logger.ErrorContext(ctx, "failed to find correlation data in context", "error", err)
```

Comment:
Should we also add an event to the current span using [recordError](https://github.com/open-telemetry/opentelemetry-go/blob/eb988d7a850f4e09763f77ed9f2ff3f02bb7a05a/sdk/trace/span.go#L518-L522)?